src/components/CytoscapeGraph/Layout/BoxLayout.ts

```diff
@@ -245,25 +207,120 @@ export default class BoxLayout {
+            boxLayoutOptions = namespaceBoxLayout;
+          }
+          break;
+      }
+
+      // Before completing work for the box type we must wait for all individual box work to complete
+      const boxNodePromises: Promise<any>[] = [];
+
+      // (2) Prepare each box node by assigning a size and running the compound layout
+      boxNodes.each(boxNode => {
+        const boxedNodes = boxNode.children();
+        const boxedElements = boxedNodes.add(boxedNodes.edgesTo(boxedNodes));
+        const boxLayout = boxedElements.layout(boxLayoutOptions);
+
+        boxNodePromises.push(
+          new Promise((resolve, _reject) => {
+            // This promise resolves when the layout actually stops.
+            this.runLayout(boxLayoutOptions.name, boxLayout).then(_response => {
+              // (2.a) get the bounding box
+              // see https://github.com/cytoscape/cytoscape.js/issues/2402
+              const boundingBox = boxNode.boundingBox();
+
+              // Save the relative positions, as we will need them later.
+              boxedNodes.each(boxedNode => {
+                boxedNode.scratch(RELATIVE_POSITION_KEY, boxedNode.relativePosition());
+              });
+
+              const backupStyles: OverridenStyles = {
+                shape: boxNode.style('shape'),
+                height: boxNode.style('height'),
+                width: boxNode.style('width')
+              };
+
+              const newStyles: OverridenStyles = {
+                shape: 'rectangle',
+                height: `${boundingBox.h}px`,
+                width: `${boundingBox.w}px`
+              };
+
+              // Saves a backup of current styles to restore them after we finish
+              boxNode.scratch(STYLES_KEY, backupStyles);
+              boxNode.addClass(BOX_NODE_CLASS);
+
+              // (2.b) Set the size
+              boxNode.style(newStyles);
+
+              resolve(true);
+            });
+          })
+        );
+      });
+
+      Promise.all(boxNodePromises).then(_results => {
+        let removedElements = this.cy.collection();
+        let syntheticEdges = this.cy.collection();
+
+        // (2.c) Add synthetic edges for every edge that touches a child node.
+        const boxedNodes = boxNodes.children();
+        for (const boxedNode of boxedNodes) {
+          for (const edge of boxedNode.connectedEdges()) {
+            const syntheticEdge = this.syntheticEdgeGenerator.getEdge(edge.source(), edge.target());
+            if (syntheticEdge) {
+              syntheticEdges = syntheticEdges.add(this.cy.add(syntheticEdge));
+            }
+          }
+        }
+        // (2.d) Remove all child nodes from parents (and their edges).
+        removedElements = removedElements.add(this.cy.remove(boxedNodes));
+
+        resolve({ boxNodes: boxNodes, syntheticEdges: syntheticEdges, removedElements: removedElements });
+      });
+    });
+  }
+
+  async runLayout(layoutName, layout): Promise<any> {
```

Comment:
minor: since you are not using `await` inside this func, you can remove the `async` keyword.

Reply:
The async keyword is still useful because it ensures a promise is returned.   Even if the function returns a non-promise it is wrapped in a resolved promise.  I know I needed this in at least one of the functions, but either way it doesn't really hurt anything.  I'll keep it for now.